### PR TITLE
Initial the XGBoost Operator with Kubebuilder 2.0

### PR DESCRIPTION
**Pull request by merlintang**
# what we proposed in this PR 
build the xgboost operator project based on kubebuilder 2.0
specific xgboost job and status 

# how to test this PR
local test

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/xgboost-operator/16)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,64 @@
+
+# Image URL to use all building/pushing image targets
+IMG ?= controller:latest
```
Comment (terrytangyuan) [inline Makefile]: Update image tag here

File: .travis.yml [hunk 2]
```diff
@@ -9,10 +9,10 @@ install:
   # get coveralls.io support
   - go get github.com/mattn/goveralls
   # gometalinter is deprecated; migrating to golangci-lint. See https://github.com/alecthomas/gometalinter/issues/590.
-  - curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh -s -- -b $(go env GOPATH)/bin v1.16.0
-
+  # - curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh -s -- -b $(go env GOPATH)/bin v1.16.0
```
Comment (terrytangyuan) [inline .travis.yml]: Let's create an issue to track this so we don't forget to turn this back on later

**Comment by terrytangyuan** [comment]
Could you also rebase since there's conflict introduced from https://github.com/kubeflow/xgboost-operator/pull/17?

**Comment by merlintang** [comment]
close this one, since we decide to use kubebuilder 1.0.8

**Pull request closed without merging**